stl/inc/regex

```diff
@@ -193,19 +193,29 @@ struct _Char_traits_lt {
     }
 };
 
+// signed char and other unsigned integral types are supported as an extension.
+template <class _Ty>
+constexpr bool _Is_predefined_char_like_type = _Is_character<_Ty>::value || is_unsigned_v<_Ty>;
+
 // library-provided char_traits::eq behaves like equal_to<_Elem>
-// TRANSITION: This should not be activated for user-defined specializations of char_traits
 template <class _Elem>
 constexpr bool _Can_memcmp_elements_with_pred<_Elem, _Elem, _Char_traits_eq<char_traits<_Elem>>> =
-    _Can_memcmp_elements<_Elem, _Elem>;
+    _Is_predefined_char_like_type<_Elem> && _Can_memcmp_elements<_Elem, _Elem>;

```

Comment:
Good catch! i thought the issue is limited to methods of `basic_string`/`basic_string_view` and `find_first_of` algorithm.